languages/elixir/exercises/concept/pattern-matching/.docs/instructions.md

```diff
@@ -0,0 +1,44 @@
+In this exercise you will convert the output of two different date functions into strings.
```

Comment:
There currently is no theme/story for this exercise, which [all Concept Exercises should have](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsinstructionsmd). The theme/story will help make the learning experience more fun for the student. The [Elixir basics exercise](https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/basics/.docs/instructions.md) has a nice example of a (simple) theme. If you find it hard to come up with a story, I'd suggest asking other people for their suggestions. I'm sure a nice theme can be found.

Reply:
One thing you could do is to look at how other tracks have handled introducing pattern matching. I think that at the moment, there is only one other exercise: [F#'s pattern-matching exercise](https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/pattern-matching). Maybe that can be used as inspiration?

Reply:
To be perfectly honest, I didn't find F#'s "story" particularly inspiring. I think we could come up with a concept focusing on guard clauses that would be a stronger parallel to the F# concept because I think an Elixir implementation would rely more on guard clauses than pattern matching for that. I think there will always be a conflict to make exercises that prioritize either being dependency-free OR are interesting.  It feels like trying to have a bake-off competition when you limit the recipes to a single ingredient.

Reply:
> To be perfectly honest, I didn't find F#'s pattern-match "story" particularly inspiring. I think we could come up with a concept focusing on guard clauses that would be a stronger parallel to that particular F# exercise because I think an Elixir implementation would rely more on guard clauses than pattern matching for that.

Sure, whatever works for Elixir. I'm just saying that this exercise currently is lacking a theme, which [is required for instructions.md files](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsinstructionsmd). The theme does not have to be very extensive, and I certainly did not want to suggest you'd just copy F#'s theme. I presented that as inspiration. 

> I think there will always be a conflict to make exercises that prioritize either being dependency-free OR are interesting. It feels like trying to have a bake-off competition when you limit the recipes to a single ingredient.

I'm not sure I fully understand how this comment applies to my original comment. Maybe it is good to remember what Concept Exercises are for: teaching Concepts to students. Making them interesting will probably be harder than v2 exercises as their scope will be more limited, but the limited scope will enable key v3 features: less diverse solutions (great for representers), less chance of students getting stuck, and just a general focus on a Concept. The existing v2 exercises will all become Practice Exercises in v3 and those exercises will probably be more interesting in terms of different approaches and such.